File: 6kyu/(6 kyu) Decode the Morse code/(6 kyu) Decode the Morse code.ts
export function decodeMorse(morseCode: string): string {
    return morseCode
    .trim()
    .split('   ')
    .map((str: string) =>
    str
    .split(' ')
        .map((s: string) => MORSE_CODE[s])
        .join(''),
    )
    .join(' ');
}

File: 6kyu/(6 kyu) Decode the Morse code/(6 kyu) Decode the Morse code.rs
impl MorseDecoder {

    fn decode_morse(&self, encoded: &str) -> String {
        encoded
            .trim()
            .split("   ")
            .map(|x| x.split(' ')
                      .filter_map(|y| { self.morse_code.get(y) })
                      .cloned()
                      .collect())
            .collect::<Vec<String>>()
            .join(" ")
    }
    
}
